integration-tests/guice3/pom.xml

```diff
@@ -123,4 +123,16 @@
         </dependency>
 	</dependencies>
 
+	<profiles>
+		<profile>
+			<id>jdk16</id>
+			<activation>
+				<jdk>[16,)</jdk>
+			</activation>
+			<properties>
+				<surefire.argLine>--illegal-access=permit</surefire.argLine>
```

Comment:
nit: add a comment as to why this is needed (to give us a hint when to remove it in the future)